File: ios/AccordLavenderMarriage/TikTokAnalyticsBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TikTokAnalyticsBridge, NSObject)

RCT_EXTERN_METHOD(trackRegistration)
RCT_EXTERN_METHOD(trackLogin)
RCT_EXTERN_METHOD(trackCompleteTutorial)
RCT_EXTERN_METHOD(identifyUser:(NSString *)externalId email:(NSString *)email)
RCT_EXTERN_METHOD(trackStartTrial)
RCT_EXTERN_METHOD(trackSubscribe)
RCT_EXTERN_METHOD(trackPurchase:(NSString *)productId productName:(NSString *)productName price:(double)price currency:(NSString *)currency)
RCT_EXTERN_METHOD(trackAddPaymentInfo)
RCT_EXTERN_METHOD(trackViewContent:(NSString *)profileId profileName:(NSString *)profileName)
RCT_EXTERN_METHOD(trackSearch)
RCT_EXTERN_METHOD(trackGenerateLead)
RCT_EXTERN_METHOD(trackAddToWishlist:(NSString *)profileId profileName:(NSString *)profileName)
RCT_EXTERN_METHOD(trackLaunchApp)
RCT_EXTERN_METHOD(trackCustomEvent:(NSString *)eventName)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/AccordLavenderMarriage/TikTokAnalyticsBridge.swift
import Foundation
import TikTokBusinessSDK

@objc(TikTokAnalyticsBridge)
class TikTokAnalyticsBridge: NSObject {

  @objc func trackRegistration() {
    TikTokAnalytics.trackRegistration()
  }

  @objc func trackLogin() {
    TikTokAnalytics.trackLogin()
  }

  @objc func trackCompleteTutorial() {
    TikTokAnalytics.trackCompleteTutorial()
  }

  @objc func identifyUser(_ externalId: String, email: String?) {
    TikTokAnalytics.identifyUser(externalId: externalId, email: email)
  }

  @objc func trackStartTrial() {
    TikTokAnalytics.trackStartTrial()
  }

  @objc func trackSubscribe() {
    TikTokAnalytics.trackSubscribe()
  }

  @objc func trackPurchase(_ productId: String, productName: String, price: Double, currency: String) {
    TikTokAnalytics.trackPurchase(productId: productId, productName: productName, price: price, currency: currency)
  }

  @objc func trackAddPaymentInfo() {
    TikTokAnalytics.trackAddPaymentInfo()
  }

  @objc func trackViewContent(_ profileId: String, profileName: String) {
    TikTokAnalytics.trackViewContent(profileId: profileId, profileName: profileName)
  }

  @objc func trackSearch() {
    TikTokAnalytics.trackSearch()
  }

  @objc func trackGenerateLead() {
    TikTokAnalytics.trackGenerateLead()
  }

  @objc func trackAddToWishlist(_ profileId: String, profileName: String) {
    TikTokAnalytics.trackAddToWishlist(profileId: profileId, profileName: profileName)
  }

  @objc func trackLaunchApp() {
    TikTokAnalytics.trackLaunchApp()
  }

  @objc func trackCustomEvent(_ eventName: String) {
    TikTokAnalytics.trackCustomEvent(eventName: eventName)
  }
}
